IP2CR Crashes When Scanning an EC2 Instance Without an IPv6 Address

**Describe the bug**
When IP2CR goes to evaluate a given EC2 instance in the ec2 plugin, the application crashes.

Traceback:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x9e887e]

goroutine 15 [running]:
github.com/magneticstain/ip-2-cloudresource/src/plugin/ec2.EC2Plugin.SearchResources({{{{0xc00002800b, 0x9}, {0xbcf480, 0xc0001f8800}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, ...}}}, ...)
	/home/josh/Code/ip-2-cloudresource/src/plugin/ec2/ec2.go:50 +0x45e
github.com/magneticstain/ip-2-cloudresource/src/search.Search.SearchAWS({{{{0xc00002800b, 0x9}, {0xbcf480, 0xc0001f8800}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, ...}}, ...}, ...)
	/home/josh/Code/ip-2-cloudresource/src/search/search.go:100 +0x2f8
github.com/magneticstain/ip-2-cloudresource/src/search.Search.doAccountSearch({{{{0xc00002800b, 0x9}, {0xbcf480, 0xc0001f8800}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, ...}}, ...}, ...)
	/home/josh/Code/ip-2-cloudresource/src/search/search.go:142 +0x6de
github.com/magneticstain/ip-2-cloudresource/src/search.Search.runSearchWorker({{{{0xc00002800b, 0x9}, {0xbcf480, 0xc0001f8800}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, ...}}, ...}, ...)
	/home/josh/Code/ip-2-cloudresource/src/search/search.go:172 +0x61e
created by github.com/magneticstain/ip-2-cloudresource/src/search.Search.InitSearch in goroutine 1
	/home/josh/Code/ip-2-cloudresource/src/search/search.go:223 +0xc69
```

**To Reproduce**
Steps to reproduce the behavior:

1. How was IP2CR installed? local
2. What CLI parameters were used? `
3. What data was provided?
```
"--ipaddr=107.20.24.235", "--verbose"
```

**Expected behavior**
IP2CR evaluates the instance properly without crashing.

**Screenshots**
If applicable, add screenshots to help explain your problem.

**Environment:**

- OS: [e.g. `Ubuntu 22.04`, `Mac OS 13/Ventura`] Ubuntu 22.04
- GO version (`go version`): [e.g. `go version go1.20.5 linux/amd64`] `1.21.3`
- IP2CR version (`ip2cr --help`]: [e.g. `v0.0.1`] v1.0.5

**Additional context**
Add any other context about the problem here.

--- a/README.md
+++ b/README.md
@@ -131,7 +131,7 @@ Instead, this will present itself as a fatal error within IP2CR, e.g.:
 ```
 INFO[0000] starting IP-2-CloudResource                  
 DEBU[0000] generating AWS connection                    
-FATA[0000] assume role with MFA enabled, but AssumeRoleTokenProvider session option not set.
+FATA[0000] assume role with MFA enabled, but AssumeRoleTokenProvider session option not set. 
 ```
 
 #### Workaround

--- a/src/plugin/ec2/ec2.go
+++ b/src/plugin/ec2/ec2.go
@@ -46,8 +46,16 @@ func (ec2p EC2Plugin) SearchResources(tgtIP string) (generalResource.Resource, e
 	for _, ec2Reservation := range ec2Resources {
 		// unpack instances from reservation
 		for _, instance := range ec2Reservation.Instances {
-			publicIPv4Addr = *instance.PublicIpAddress
-			IPv6Addr = *instance.Ipv6Address
+			evalAddrPtr := func(addrPtr *string) string {
+				addr := ""
+				if addrPtr != nil {
+					addr = *addrPtr
+				}
+
+				return addr
+			}
+			publicIPv4Addr = evalAddrPtr(instance.PublicIpAddress)
+			IPv6Addr = evalAddrPtr(instance.Ipv6Address)
 
 			if publicIPv4Addr == tgtIP || IPv6Addr == tgtIP {
 				matchingResource.RID = *instance.InstanceId // for some reason, the EC2 Instance object doesn't contain the ARN of the instance :/

